guides-markdown/wallets.md

```diff
@@ -0,0 +1,403 @@
+
+```
+~$ curl http://127.0.0.1:18332/wallet/watchonlywallet/import -X POST -d '{"account":"default"
+,"privateKey":"cNZfR3NhQ9oCP3pTjvPZETUuTW
+Zo2k6EXtfczvbWyv7FdjMhppvJ"}'
+{
+  "error": {
+    "type": "Error",
+    "message": "Cannot import privkey into watch-only wallet."
+  }
+}
+~$ curl http://127.0.0.1:18332/wallet/privatekeywallet/import -X POST -d '{"account":"default","publicKey":"02f4f200cb9391f8bbcc0a35e1f654b9b993b214a04ae7efd0313f4d4bf3d95745"}'{
+  "error": {
+    "type": "Error",
+    "message": "Cannot import pubkey into non watch-only wallet."
+  }
+}
+```
+
+### API Authentication
+
+Bcoin can run as a server and allow you to interact with your wallets via a [REST API](http://bcoin.io/api-docs/index.html?shell--curl#wallet). It also allows you protect wallets from unauthenticated requests by running the server with the `wallet-auth` configuration option. Each wallet you create will have a `token` value that must be passed in with each request. Tokens, like accounts and keys, can also be deterministically generated using your HD seed. This means you can change the token on a wallet as often as you'd like.
+
+### Recovery
+
+By using the established standards mentioned above, bcoin allows one to easily restore or transfer their entire wallet to different wallet implementations. With only that twelve word mnemonic, one can fully recover their wallet to a fresh instance of bcoin. Furthermore, you can recreate the wallet with only your seed on any other software that properly implements BIP33, BIP39, and BIP44 mentioned above, like the [Trezor](https://trezor.io/) hardware wallet.
+
+## Examples
+
+Enough chit chat, let's get down to business on how to create wallets, accounts, and keys with bcoin.
+
+### NodeJS
+
+Below is a demo using javascript to instatiate a wallet and output important data and keys. Since bcoin is modular, you can easily use just the wallet functionality as I've done here.
+
+#### Setup
+
+With the [bcoin npm package](https://www.npmjs.com/package/bcoin) installed, we'll first import bcoin and open a wallet database in memory.
+
+```javascript
+//import the bcoin module and set it to testnet
+const bcoin = require('bcoin').set('testnet');
+const WalletDB = bcoin.walletdb;
+const WalletKey = bcoin.walletkey;
+const KeyRing = bcoin.keyring;
+const Mnemonic = bcoin.hd.Mnemonic;
+const HD = bcoin.hd;
+
+walletExample().catch(console.error.bind(console));
+
+async function walletExample() {
+	//for demonstration purposes, we'll be creating a temporary wallet in memory
+	const wdb = new WalletDB({ db: 'memory' });
+	await wdb.open();
+```
+
+#### Creating a Wallet
+
+Creating a wallet in our database takes only one line.
+
+```javascript
+	//creates and returns a Wallet object from scratch using a random master key and default options
```

Comment:
can you make the first level indent flush against the left. The markdown comes out very far to the right.

Reply:
The rest underneath until the shell command also need to the indent to be decreased.

Reply:
Gotcha, the indent was just copied over along with the code from my editor, I'll make it flush to the left.